http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -59,12 +59,27 @@ public Exception decode(String methodKey, Response response) {
                 try {
                     SerializableError error = MAPPER.readValue(bodyAsString, SerializableError.class);
                     Class<? extends Exception> exceptionClass = error.getExceptionClass();
-                    Exception exception = exceptionClass.getConstructor(String.class).newInstance(error.getMessage());
+
+                    // Construct remote exception
+                    Exception remoteException =
+                            exceptionClass.getConstructor(String.class).newInstance(error.getMessage());
```

Comment:
we probably ought to do this construction and the one on line 74 in a utility method that safely tries to construct each of these and failing that just produces a RuntimeException holding data and shoves the classname of the uninstantiable exception into the error message.